### ISSUE TITLE
Ruby install fails due to postgresql mkdir call

### ISSUE THREAD

**Issue by efatsi** [asdf-vm/asdf-ruby#273]
I'm seeing this issue on attempts to install multiple Ruby versions:
```
$ asdf install ruby 3.1.2
Downloading openssl-1.1.1n.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/40dceb51a4f6a5275bde0e6bf20ef4b91bfc32ed57c0552e2e8e15463372b17a
Installing openssl-1.1.1n...
Installed openssl-1.1.1n to /Users/efatsi/.asdf/installs/ruby/3.1.2

Downloading ruby-3.1.2.tar.gz...
-> https://cache.ruby-lang.org/pub/ruby/3.1/ruby-3.1.2.tar.gz
Installing ruby-3.1.2...
mkdir: /usr/local/opt/postgresql@11/lib: Permission denied

BUILD FAILED (macOS 12.2.1 using ruby-build 20220426)


Inspect or clean up the working tree at /var/folders/w7/_xmx5jj50p90jn3r2d_1dvcc0000gn/T/ruby-build.20220525113213.58166.BaY6fq
Results logged to /var/folders/w7/_xmx5jj50p90jn3r2d_1dvcc0000gn/T/ruby-build.20220525113213.58166.log
```

here's the full log that's spat out - https://gist.github.com/efatsi/1f78d02a71e6b9ca1bfefae13283dedd.

It's surprising to me that there's anything to do with `postgresql` in the process of building ruby. It's also the case that `/usr/local/opt` is not a directory on my machine, and I'm not sure where the path `/usr/local/opt/postgresql@11/lib` is coming from.

**Comment by pboling** [comment]
I'd guess that it is coming from legacy dot files config you brought over from another machine.  It is almost certainly something arising from your local machine's config, and not this project in any way.

The path `/usr/local/opt/postgresql@11/lib` is from a Homebrew install of PostgreSQL on an **_Intel_** Mac.  Did you get a new **_Apple Silicon_** Mac and bring over the system from the **_Intel_** Mac?  If so you have to uninstall and reinstall Homebrew, as the Intel and Apple Silicon packages are not entirely compatible.

@Stratus3D this should be closed.

**Comment by monfresh** [comment]
I agree with Peter's assessment above. One of the most common reasons for Ruby installation issues like the one you're seeing is specifying env vars such as `LDFLAGS` globally in your shell startup file (such as `~/.zshrc` and `~/.bash_profile`). If you open one of those files, I am willing to bet that you are referencing `/usr/local/opt/postgresql@11/lib` in there. Another way to check which env vars are being set when you open a new terminal session is to run `env` and then do a search for "postgres".

ENV vars that can affect Ruby compilation should never ever be set globally in a shell startup file because they can also affect the installation of other tools. And they might only apply to certain situations. As Peter mentioned above, the `/usr/local/opt` directory is where Homebrew installs things on the Intel architecture, but if you're trying to install Ruby on an Apple Silicon Mac, setting an env var that points to `/usr/local` will cause issues. 

In your case, the issue is that you're pointing to a nonexistent directory, but it's also common for people to install Homebrew both in `/opt/homebrew` (the location for the arm64 architecture that Apple SIlicon Macs use), and in `/usr/local` by running the terminal in Rosetta mode, which simulates the Intel architecture. So even if you did have an Intel-based Homebrew installation (via Rosetta), you would still run into issues because you can't mix and match the arm64 and Intel architectures.

TL;DR Don't blindly add things to your shell startup file that you don't fully understand. If you know what you are doing, and you know for sure that you need to set an env var, it should be done directly on the command line while installing Ruby.

**Comment by Stratus3D** [comment]
Closing as I don't think there is anything we can do in asdf-ruby for this.
